### PR TITLE
Fix pod_mutation_hook for 1.10.13

### PR DESCRIPTION
**Pull request by dimberman**
We missed backporting a few feature overrides for the legacy pod_mutation_hook. This PR will add those overrides to ensure users have full backwards compatibility.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by ashb — APPROVED** [review]

**Review by kaxil — COMMENTED** [review]
Can you please update PR title and add a short PR description to describe what this PR fixes

**Comment by kaxil** [comment]
I have restarted the tests too, they were cancelled for some reason

**Comment by kaxil** [comment]
Tests are failing

File: airflow/kubernetes/pod_launcher.py [hunk 1]
```diff
@@ -103,6 +103,7 @@ def _mutate_pod_backcompat(pod):
             dummy_pod = dummy_pod.to_v1_kubernetes_pod()
 
             new_pod = PodGenerator.reconcile_pods(pod, dummy_pod)
+            print(new_pod)
```
Comment (kaxil) [inline airflow/kubernetes/pod_launcher.py]:
```suggestion
```

**Review by kaxil — APPROVED** [review]

**Pull request merged**